es72x/src/main/scala/tech/beshu/ror/es/dlsfls/RoleIndexSearcherWrapper.scala

```diff
@@ -0,0 +1,139 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.es.dlsfls
+
+import java.io.IOException
+
+import cats.data.StateT
+import cats.implicits._
+import com.google.common.base.Strings
+import eu.timepit.refined.types.string.NonEmptyString
+import org.apache.logging.log4j.scala.Logging
+import org.apache.lucene.index.DirectoryReader
+import org.apache.lucene.search.{BooleanClause, BooleanQuery, ConstantScoreQuery}
+import org.elasticsearch.ExceptionsHelper
+import org.elasticsearch.common.logging.LoggerMessageFormat
+import org.elasticsearch.common.util.concurrent.ThreadContext
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.json.JsonXContent
+import org.elasticsearch.index.IndexService
+import org.elasticsearch.index.shard.{IndexSearcherWrapper, ShardId, ShardUtils}
+import tech.beshu.ror.Constants
+import tech.beshu.ror.accesscontrol.headerValues.transientFieldsFromHeaderValue
+import tech.beshu.ror.utils.FilterTransient
+
+import scala.util.{Failure, Success, Try}
+
+class RoleIndexSearcherWrapper(indexService: IndexService) extends IndexSearcherWrapper with Logging {
+
+  override def wrap(reader: DirectoryReader): DirectoryReader = {
+    val threadContext: ThreadContext = indexService.getThreadPool.getThreadContext
+    val result = for {
+      _ <- prepareDocumentFieldReader(threadContext)
+      newReader <- prepareDocumentFilterReader(threadContext, indexService)
+    } yield newReader
+    result.run(reader).get._2
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://app.codacy.com/manual/scarduzio/elasticsearch-readonlyrest-plugin/pullRequest?prid=4803411)